Sources/HTTP/HTTPVersion.swift

```diff
@@ -29,12 +29,44 @@ extension HTTPVersion : Hashable {
     public static func == (lhs: HTTPVersion, rhs: HTTPVersion) -> Bool {
         return lhs.hashValue == rhs.hashValue
     }
-
+    
+    public static func != (lhs: HTTPVersion, rhs: HTTPVersion) -> Bool {
+        return lhs.hashValue != rhs.hashValue
+    }
+    
     public static func ~= (match: HTTPVersion, version: HTTPVersion) -> Bool {
         return match == version
     }
 }
 
+extension HTTPVersion : Comparable {
+    
+    public static func > (lhs: HTTPVersion, rhs: HTTPVersion) -> Bool {
+        if lhs.major != rhs.major {
+            return lhs.major > rhs.major
+        } else {
+            return lhs.minor > rhs.minor
+        }
+    }
+    
+    public static func >= (lhs: HTTPVersion, rhs: HTTPVersion) -> Bool {
+        return lhs > rhs || lhs == rhs
+    }
+    
+    public static func < (lhs: HTTPVersion, rhs: HTTPVersion) -> Bool {
+        if lhs.major != rhs.major {
+            return lhs.major < rhs.major
+        } else {
+            return lhs.minor < rhs.minor
+        }
+    }
+    
+    public static func <= (lhs: HTTPVersion, rhs: HTTPVersion) -> Bool {
+        return lhs < rhs || lhs == rhs
+    }
+
+}
+
```

Comment:
Agreed - looking at the implementation in the stdlib:
https://github.com/apple/swift/blob/767e8f31ac8ab8186102dc95fe5916bb2d29bbb5/stdlib/public/core/Comparable.swift
there's no advantage to implementing these by hand. I'll update to remove.